### PR TITLE
Merge March 8 security tests into their main class

### PR DESCRIPTION
**Pull request by Kevin-CB**
This PR aims 2 goals:
* Merge back separate tests file added during the 8 March security fix to their main test class. (they've been added in separate file to avoid conflicts during merges)
  * [PluginManagerTest](https://github.com/jenkinsci/jenkins/commit/52c58e2446b16a007ee895a37a60e7e5bd739407)
  * [DirectoryBrowserSupportTest](https://github.com/jenkinsci/jenkins/commit/db05ca7a1f80c9a7cbf3f4dbae65a91b5a6a151c)
  * [ComputerSetTest & ExecutorTest](https://github.com/jenkinsci/jenkins/commit/8a9f621a325db5752674b828a9a82e6aebf13957)
  * [SearchTest](https://github.com/jenkinsci/jenkins/commit/6748ee877d2e5b47da2642877ac9431ed3e0fe50)
* Fix one of the newly added tests that turned out to be falky
  * [adjust PluginManagerTest](https://github.com/jenkinsci/jenkins/commit/91f23fb3d6ae35d7c5ba1b6e71c7bb155f25456f)

### Testing done

Modified tests successfully run on Java 11 and Java 17 on Ubuntu 22.04, RHEL 8.7, and Debian testing

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7700"></a>

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]

**Review by yaroslavafenkin — APPROVED** [review]

File: test/src/test/java/hudson/PluginManagerTest.java [hunk 1]
```diff
@@ -711,4 +735,101 @@ public void installNecessaryPluginsTest() throws Exception {
         // Redirect reported 404 before bug was fixed
         assertEquals("Bad response for installNecessaryPlugins", 200, response.statusCode());
     }
+
+    @Test
+    @Issue("SECURITY-2823")
+    public void verifyUploadedPluginPermission() throws Exception {
```
Comment (timja) [inline test/src/test/java/hudson/PluginManagerTest.java:741]: can we ignore this test until it is stabilised please?

I can reproduce it 100% with:
```
mvn test -Dtest=hudson.PluginManagerTest#verifyUploadedPluginPermission
```

It passes in IntelliJ though

(have tried a bit but haven't figured it out)

Comment (timja) [reply test/src/test/java/hudson/PluginManagerTest.java:741]: (although this test does pass for me on master weirdly although failing I think most of the time on CI)

Comment (MarkEWaite) [reply test/src/test/java/hudson/PluginManagerTest.java:741]: Tests pass for me locally with Java 11 on RHEL 8 with the command:

```
mvn clean -DforkCount=1C -Dtest=PluginManagerTest,DirectoryBrowserSupportTest,ComputerSetTest,ExecutorTest,SearchTest clean verify
```

**Comment by Kevin-CB** [comment]
Apparently on Java 17, `File.createTempFile()` ignore the Java system property `java.io.tmpdir` while `FileS.createTempFile()` don't ignore it. Resulting in our hpi file created in a different folder than the one we're checking.

**Review by MarkEWaite — APPROVED** [review]
`PluginManagerTest` is now passing with Java 17 on RHEL 8, Java 17 on Ubuntu 22.04, and Java 17 on Debian testing.  Thanks very much!

**Comment by MarkEWaite** [comment]
I recommend that we skip the 24 hour merge window for this and instead merge it as soon as CI passes.

**Comment by NotMyFault** [comment]
+1

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**